snmplib/transports/snmpUDPIPv4BaseDomain.c

```diff
@@ -301,6 +301,17 @@ netsnmp_udpipv4base_transport_with_source(const struct netsnmp_ep *ep,
     if (NULL == bind_addr)
         return t;
 
+    /* for Linux VRF Traps we try to bind the iface if clientaddr is not set */
```

Comment:
This line is only reached if bind_addr != NULL. If clientaddr has not been set then src_addr == NULL. So this comment is not clear to me.

Reply:
we do have a bind_addr (sink) but we also have an iface we need to bind on the socket.  Otherwise, at least for linux, we will not send the trap out the right interface.